packages/@webex/plugin-cc/src/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable */
+import { Signal } from '../Signal';
+import { WebexSDK, SubscribeRequest, HTTP_METHODS, WelcomeResponse } from '../../../types';
+import { SUBSCRIBE_API, WCC_API_GATEWAY } from '../../constants';
+import { SubscribeResponse } from '../../config/types';
+import workerScript from './keepalive.worker';
+
+export class WebSocketManager {
+  readonly onMessage: Signal.WithData<string>;
+  private readonly onMessageSend: Signal.Send<string>;
+  readonly onSocketClose: Signal.Empty;
+  private readonly onSocketCloseSend: Signal.SendEmpty;
+  private websocket: WebSocket;
+  shouldReconnect: boolean;
+  isSocketClosed: boolean;
+  private isWelcomeReceived: boolean;
+  private url: string | null = null;
+  private forceCloseWebSocketOnTimeout: boolean;
+  private isConnectionLost: boolean;
+  private webex: WebexSDK;
+  private welcomePromiseResolve: ((value: WelcomeResponse | PromiseLike<WelcomeResponse>) => void) | null = null;
+
+  private keepaliveWorker: Worker;
+
+  constructor(options: {webex: WebexSDK}) {
+    const {webex} = options;
+    this.webex = webex;
+    const { send, signal } = Signal.create.withData<string>();
+    this.onMessage = signal;
+    this.onMessageSend = send;
+
+    const socketCloseSignal = Signal.create.empty();
+    this.onSocketClose = socketCloseSignal.signal;
+    this.onSocketCloseSend = socketCloseSignal.send;
+    this.shouldReconnect = true;
+    this.websocket = {} as WebSocket;
+    this.isSocketClosed = false;
+    this.isWelcomeReceived = false;
+    this.forceCloseWebSocketOnTimeout = false;
+    this.isConnectionLost = false;
+
+    const workerScriptBlob = new Blob([workerScript], { type: 'application/javascript' });
```

Comment:
Note - 
This is a workaround, currently I was not able to find any way to import the script the same way we were doing in agent desktop (as that is a node server).
However this workaround can be improved (later) wherein we read from the .js script file rather than exporting the string.